api/v1beta2/awsmachine_types.go

```diff
@@ -197,6 +197,11 @@ type AWSMachineSpec struct {
 	// CapacityReservationID specifies the target Capacity Reservation into which the instance should be launched.
 	// +optional
 	CapacityReservationID *string `json:"capacityReservationId,omitempty"`
+
+	// UseCapacityBlock enables usage of pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	// If enabled, CapacityReservationID must be specified to identify the target reservation.
+	// +optional
+	UseCapacityBlock *bool `json:"useCapacityBlock,omitempty"`
```

Comment:
Why is this needed, if all it is doing is enabling the capacity reservation ID field?

Reply:
we need this to use for AWS capacity blocks for ml.